File: src/GradeMe3/Evaluations/Evaluation.cs
using System;
using GradeMe3.Projects;
using GradeMe3.Students;
using Newtonsoft.Json;

namespace GradeMe3.Evaluations
{
    public class Evaluation
    {
        public string Id { get; set; }
        public DateTime Time { get; set; }
        public string EvaluatorStudentId { get; set; }
        public string ProjectId { get; set; }
        public string EvaluatedStudentId { get; set; }

        public int Grade { get; set; }
        public string Justification { get; set; }

        [JsonIgnore]
        public Student EvaluatorStudent { get; set; }        
        [JsonIgnore]
        public Student EvaluatedStudent { get; set; }        
        [JsonIgnore]
        public Project Project { get; set; }        
    }
}

File: src/server/entities/Evaluation.ts
import {Column, Entity, JoinColumn, ManyToOne, OneToMany, PrimaryColumn} from 'typeorm';
import {Student} from './Student';
import {Project} from './Project';
import {EvaluationApproval} from './EvalutationApproval';

@Entity()
export class Evaluation {
    @PrimaryColumn()
    public id: string;
    @ManyToOne(type => Student, student => student.evaluationsBy)
    public evaluatorStudent: Student;
    @ManyToOne(type => Student, student => student.evaluationsTo)
    public evaluatedStudent: Student;
    @Column('double')
    public grade: number;
    @Column()
    public justification: string;
    @ManyToOne(type => Project, project => project.evaluations)
    public project: Project;
    @Column('datetime')
    public time: Date;
    @OneToMany(type => EvaluationApproval, approval => approval.evaluation)
    public approvals: EvaluationApproval[];
}